vdp/pipeline/v1beta/component_definition.proto

```diff
@@ -69,15 +69,64 @@ message ComponentDefinition {
     RELEASE_STAGE_GA = 5;
   }
 
-  // Defines the type of task the component will perform.
-  ComponentType type = 1 [(google.api.field_behavior) = OUTPUT_ONLY];
-  // The component definition details.
-  oneof definition {
-    // operator definition detail
-    OperatorDefinition operator_definition = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
-    // connector definition detail
-    ConnectorDefinition connector_definition = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Spec represents a specification data model.
+  message Spec {
+    // Component specification.
+    google.protobuf.Struct component_specification = 3 [(google.api.field_behavior) = REQUIRED];
+    // Data specifications.
+    // The key represents the task, and the value is the corresponding data_specification.
+    map<string, DataSpecification> data_specifications = 5 [(google.api.field_behavior) = REQUIRED];
   }
+
+  // The name of the component definition, defined by its ID.
+  // - Format: `component-definitions/{id}
+  string name = 1 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Component definition UUID.
+  string uid = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Component definition resource ID (used in `name` as the last segment). This
+  // conforms to RFC-1034, which restricts to letters, numbers, and hyphen,
+  // with the first character a letter, the last a letter or a number, and a 63
+  // character maximum.
+  string id = 3 [(google.api.field_behavior) = IMMUTABLE];
+  // Component definition title.
+  string title = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Component definition documentation URL.
+  string documentation_url = 5 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // ConnecComponenttor definition icon. This is a path that's relative to the root of
+  // the connector implementation (see `source_url`) and that allows clients
+  // frontend applications to pull and locate the icons.
+  string icon = 6 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Component definition specification.
+  Spec spec = 7 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Component definition type.
+  ComponentType type = 8 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Component definition tombstone. If true, this configuration is permanently
+  // off. Otherwise, the configuration is active.
+  bool tombstone = 9 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // The public flag determines whether this connector definition is available
+  // to all workspaces.
+  bool public = 10 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Connector definition custom flag, i.e., whether this is a custom
+  // connector definition.
+  bool custom = 11 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // field 12 is reserved for 'icon_url'.
+  reserved 12;
```

Comment:
❓ Perhaps we don't need this anymore, since it's part of the definitions we're removing.